promises/es7Promises.js

```diff
@@ -0,0 +1,52 @@
+/* ES7 */
+const isMomHappy = true;
+
+// Promise
+const willIGetNewPhone = new Promise(
+    (resolve, reject) => {
+        if (isMomHappy) {
+            const phone = {
+                brand: 'Samsung',
+                color: 'black'
+            };
+            resolve(phone);
+        } else {
+            const reason = new Error('mom is not happy');
+            reject(reason);
+        }
+
+    }
+);
+
+// 2nd promise
+// Whenever you need to return a promise in a function, you prepend async to that function. 
+async function showOff(phone) {
+    return new Promise(
+        (resolve, reject) => {
+            var message = 'Hey friend, I have a new ' +
+                phone.color + ' ' + phone.brand + ' phone';
+
+            resolve(message);
+        }
+    );
+};
+
+// call our promise
+async function askMom() {
+    try {
+        console.log('before asking Mom');
+
+        let phone = await willIGetNewPhone;//Whenever you need to call a promise, you prepend with await
```

Comment:
alternatively we can wait for all the promises to resolve using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
but good async and await example

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function
Do not confuse await for Promise.all
In add1, execution suspends 2 seconds for the first await, and then again another 2 seconds for the second await. The second timer is not created until the first has already fired. In add2,  both timers are created, and then both are awaited. This leads it to resolve in 2 rather than 4 seconds, because the timers are running concurrently. But both of the await calls are still run in series, not in parallel: this is not some automatic application of Promise.all. If you wish to await two or more promises in parallel, you must still use Promise.all.